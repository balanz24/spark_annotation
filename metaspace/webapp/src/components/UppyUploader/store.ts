// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from '@vue/composition-api'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach(listener => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
