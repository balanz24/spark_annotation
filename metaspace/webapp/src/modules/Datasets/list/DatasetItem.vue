<template>
  <div
    v-if="!deferRender"
    class="dataset-item border border-solid border-gray-200 leading-5"
  >
    <div
      v-if="isOpticalImageSupported"
      class="opt-image--container"
    >
      <dataset-thumbnail
        :dataset="dataset"
        :editable="canEditOpticalImage"
      />
    </div>

    <DatasetItemMetadata
      :dataset="dataset"
      :metadata="metadata"
      :hide-group-menu="hideGroupMenu"
    />

    <DatasetItemActions
      :dataset="dataset"
      :idx="idx"
      :metadata="metadata"
      :additional-settings="additionalSettings"
      :current-user="currentUser"
    />
  </div>
</template>

<script>
import DatasetThumbnail from './DatasetThumbnail.vue'
import { get } from 'lodash-es'
import { mdTypeSupportsOpticalImages } from '../../../lib/util'
import safeJsonParse from '../../../lib/safeJsonParse'
import { plural } from '../../../lib/vueFilters'
import { defaultMetadataType } from '../../../lib/metadataRegistry'
import DatasetItemMetadata from './DatasetItemMetadata'
import DatasetItemActions from './DatasetItemActions'

export default {
  name: 'DatasetItem',
  components: {
    DatasetThumbnail,
    DatasetItemMetadata,
    DatasetItemActions,
  },
  filters: {
    plural,
  },
  props: ['dataset', 'currentUser', 'idx', 'hideGroupMenu'],
  data() {
    return {
      deferRender: this.idx >= 20,
    }
  },

  computed: {
    isOpticalImageSupported() {
      return mdTypeSupportsOpticalImages(get(this.metadata, 'Data_Type') || defaultMetadataType)
    },

    metadata() {
      const datasetMetadataExternals = {
        Submitter: this.dataset.submitter,
        PI: this.dataset.principalInvestigator,
        Group: this.dataset.group,
        Projects: this.dataset.projects,
      }
      return Object.assign(safeJsonParse(this.dataset.metadataJson), datasetMetadataExternals)
    },

    additionalSettings() {
      try {
        const configJson = JSON.parse(this.dataset.configJson)
        return configJson
      } catch (e) {
        return {}
      }
    },

    canEditOpticalImage() {
      return this.currentUser?.role === 'admin'
        // Only allow editing after annotation has finished, as it requires ion images for alignment
        || (this.dataset.canEdit && this.dataset.status === 'FINISHED')
    },
  },
  async created() {
    // Defer rendering of most elements until after the first render, so that the page becomes interactive sooner
    const delayFrames = Math.floor(this.idx / 10)
    try {
      for (let i = 0; i < delayFrames; i++) {
        await new Promise(resolve => requestAnimationFrame(resolve))
      }
    } catch (err) { /* Browser/test doesn't support requestAnimationFrame? */ }
    this.deferRender = false
  },
}
</script>

<style lang="scss">
  .opt-image--container {
    padding: 10px 0 10px 10px;
    margin: 0px;
    flex: none;
  }

 .dataset-item {
   border-radius: 5px;
   min-height: 120px;
   min-width: 600px;
   max-width: 950px;
   margin: 3px;
   padding: 0px;
   display: flex;
   flex-direction: row;
   justify-content: space-between;
   transition: 0.2s cubic-bezier(.4, 0, .2, 1);
   transition-property: box-shadow;
 }

 .dataset-item:hover {
   box-shadow: 0 2px 12px 0 rgba(0, 0, 0, 0.1);
 }

 .ds-info {
   padding: 10px;
   margin: 0px;
   flex-grow: 1;
   min-width: 0%; /* This may seem pointless, but it's necessary to prevent text overflowing: https://css-tricks.com/flexbox-truncated-text/ */
 }

 .ds-actions {
   padding: 10px 20px 10px 0px;
   margin: 0px;
   flex: none;
   display: flex;
   flex-direction: column;
 }

 .ds-actions i {
   margin-right: 5px;
 }

 .ds-add-filter {
   cursor: pointer;
   font-weight: 500;
 }

 .striped-progressbar {
   height: 14px;
   border-radius: 2px;
   margin: 3px 0;
   width: 100%;
   background-size: 30px 30px;
   background-image: linear-gradient(135deg,
     rgba(255, 255, 255, .30) 25%, transparent 25%, transparent 50%,
     rgba(255, 255, 255, .30) 50%,
     rgba(255, 255, 255, .30) 75%, transparent 75%, transparent);

   animation: animate-stripes 3s linear infinite;
 }

 @keyframes animate-stripes {
   0% {background-position: 0 0;} 100% {background-position: 60px 0;}
 }

 .processing {
   background-color: lightgreen;
 }

 .queued {
   background-color: lightblue;
 }

 .db-link-list {
   font-size: initial;
 }

 .ds-item-line {
   overflow: hidden;
   white-space: nowrap;
   text-overflow: ellipsis;
 }

</style>
