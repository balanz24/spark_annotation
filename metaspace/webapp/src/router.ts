/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/imzml/DatasetBrowserPage'),
  DatasetEnrichmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/enrichment/DatasetEnrichmentPage'),
  SpottingProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "SpottingProjectPage" */ './modules/SpottingProject/DashboardPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),

  // Separate bundle for design docs
  DesignTOC: () => import(/* webpackChunkName: "DesignBundle" */ './design/TOCPage.vue'),
  DesignStyleGuide: () => import(/* webpackChunkName: "DesignBundle" */ './design/StyleGuidePage.vue'),
  DesignIcons: () => import(/* webpackChunkName: "DesignBundle" */ './design/IconsPage.vue'),
  DesignComponents: () => import(/* webpackChunkName: "DesignBundle" */ './design/ComponentsPage.vue'),
  DesignForms: () => import(/* webpackChunkName: "DesignBundle" */ './design/FormsPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    {
      path: '/datasets/:dataset_id/comparison',
      name: 'datasets-comparison',
      component: asyncPages.DatasetComparisonPage,
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
    { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
    { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
    { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetEnrichmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/groups', component: asyncPages.GroupsListPage },
    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignTOC, meta: { footer: true, flex: true } },
    { path: '/design/styleguide', component: asyncPages.DesignStyleGuide, meta: { footer: true, flex: true } },
    { path: '/design/icons', component: asyncPages.DesignIcons, meta: { footer: true, flex: true } },
    { path: '/design/components', component: asyncPages.DesignComponents, meta: { footer: true, flex: true } },
    { path: '/design/forms', component: asyncPages.DesignForms, meta: { footer: true, flex: true } },

    { path: '/detectability', name: 'detectability', component: asyncPages.SpottingProjectPage },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
