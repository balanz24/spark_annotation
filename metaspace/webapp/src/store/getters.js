import { mzFilterPrecision } from '../lib/util';
import { decodeParams, decodeSettings, getLevel } from '../modules/Filters';
import config from '../lib/config';

/** For filters where empty string is a valid client-side value that must be converted to empty string for the API */
const noneToEmptyString = s => s === 'none' ? '' : s;

export default {
  filterLevel(state) {
    return getLevel(state.route.path);
  },

  filter(state) {
    return decodeParams(state.route, state.filterLists);
  },

  currentUser(state) {
    return state.currentUser;
  },

  settings(state) {
    return decodeSettings(state.route);
  },

  ftsQuery(state, getters) {
    return getters.filter.simpleQuery;
  },

  gqlAnnotationFilter(state, getters) {
    const filter = getters.filter;
    const colocalizationAlgo = getters.settings.annotationView.colocalizationAlgo;

    const f = {
      compoundQuery: filter.compoundName,
      termId: filter.term ? parseInt(filter.term, 10) : undefined,
      chemMod: noneToEmptyString(filter.chemMod),
      neutralLoss: noneToEmptyString(filter.neutralLoss),
      adduct: filter.adduct,
      annotationId: filter.annotationIds ? filter.annotationIds.join('|') : undefined,
      fdrLevel: filter.fdrLevel,
      pValue: filter.pValue,
      colocalizedWith: filter.colocalizedWith,
      // Only include colocalizationAlgo if there is another filter that uses it. Otherwise the annotations list
      // refreshes unnecessarily when changing algorithm.
      colocalizationAlgo: filter.colocalizedWith || filter.colocalizationSamples ? colocalizationAlgo : null,
      colocalizationSamples: filter.colocalizationSamples,
      offSample: filter.offSample == null ? undefined : !!filter.offSample,
    };

    if (!config.features.neutral_losses && !config.features.advanced_ds_config) {
      f.hasNeutralLoss = false;
    }

    if (!config.features.chem_mods && !config.features.advanced_ds_config) {
      f.hasChemMod = false;
    }

    if (!config.features.all_adducts) {
      f.hasHiddenAdduct = false;
    }

    if (filter.minMSM)
      f.msmScoreFilter = { min: filter.minMSM, max: 1.0 };

    if (filter.mz) {
      const mz = parseFloat(filter.mz),
        deltamz = parseFloat(mzFilterPrecision(filter.mz));
      f.mzFilter = {
        min: (mz - deltamz),
        max: (mz + deltamz)
      };
    }

    f.databaseId = filter.database

    return f;
  },

  gqlDatasetFilter(state, getters) {
    const filter = getters.filter;
    const { group, project, submitter, datasetIds, polarity,
      organism, organismPart, condition, growthConditions,
      ionisationSource, analyzerType, maldiMatrix, metadataType,
      compoundName, datasetOwner, opticalImage } = filter;
    const level = getters.filterLevel;
    const isLogged = state.currentUser && state.currentUser.id
    const hasAnnotationMatching = level === 'dataset' && compoundName ? { compoundQuery: compoundName } : undefined;
    return {
      group: datasetOwner && datasetOwner !== 'my-datasets' && isLogged ? datasetOwner : group,
      project: project,
      submitter: datasetOwner === 'my-datasets' && isLogged ? state.currentUser.id : submitter,

      // temporary workaround because of array-related bugs in apollo-client
      ids: datasetIds ? datasetIds.join("|") : null,

      organism,
      opticalImage,
      organismPart,
      condition,
      growthConditions,
      ionisationSource,
      maldiMatrix,
      analyzerType,
      polarity: polarity ? polarity.toUpperCase() : null,
      metadataType,
      hasAnnotationMatching,
    }
  },

  gqlColocalizationFilter(state, getters) {
    const { datasetIds, colocalizedWith, database, fdrLevel } = getters.filter;
    const colocalizationAlgo = getters.settings.annotationView.colocalizationAlgo;
    if (datasetIds && !datasetIds.includes('|') && colocalizedWith != null && database != null && fdrLevel != null) {
      return { colocalizedWith, colocalizationAlgo, databaseId: database, fdrLevel };
    } else {
      return null;
    }
  },
}
