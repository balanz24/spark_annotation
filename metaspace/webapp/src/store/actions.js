import apolloClient from '../api/graphqlClient';
import {fetchOptionListsQuery} from '../api/metadata';
import {decodeParams} from '../modules/Filters';
import {computed} from "@vue/composition-api";
import store from "../store/index";
export default {

  async initFilterLists(context) {
    if(context.state.filterListsLoading || context.state.filterLists != null)
      return;

    context.commit('setFilterListsLoading');

    const response = await apolloClient.query({
      query: fetchOptionListsQuery,
      fetchPolicy: 'cache-first',
    });

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds)} );

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)
    context.commit('updateFilter', filter)
  },

};
