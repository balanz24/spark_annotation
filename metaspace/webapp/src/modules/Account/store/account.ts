import { Module } from 'vuex'
import { DialogType } from '../dialogs'
import router from '../../../router'
import { RawLocation } from 'vue-router/types/router'

export interface AccountState {
  dialog: DialogType | null;
  dialogCloseRedirect: RawLocation | null;
  loginSuccessRedirect: RawLocation | null;
}

interface ShowDialogOptions {
  dialog: DialogType;
  dialogCloseRedirect?: RawLocation;
  loginSuccessRedirect?: RawLocation;
}

interface HideDialogOptions {
  dialog: DialogType;
  isLoginSuccess?: boolean;
}

const account: Module<AccountState, any> = {
  namespaced: true,
  state: {
    dialog: null,
    dialogCloseRedirect: null,
    loginSuccessRedirect: null,
  },
  mutations: {
    showDialog(state, payload: DialogType | ShowDialogOptions) {
      const options: ShowDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      state.dialog = options.dialog
      if (options.dialogCloseRedirect != null) {
        state.dialogCloseRedirect = options.dialogCloseRedirect
      }
      if (options.loginSuccessRedirect != null) {
        state.loginSuccessRedirect = options.loginSuccessRedirect
      }
    },
    hideDialog(state, payload: DialogType | HideDialogOptions) {
      const options: HideDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      if (state.dialog === options.dialog) {
        state.dialog = null
        if (options.isLoginSuccess && state.loginSuccessRedirect != null) {
          router.push(state.loginSuccessRedirect)
        } else if (state.dialogCloseRedirect != null) {
          router.push(state.dialogCloseRedirect)
        }

        state.dialogCloseRedirect = null
        state.loginSuccessRedirect = null
      }
    },
  },
}

export default account
