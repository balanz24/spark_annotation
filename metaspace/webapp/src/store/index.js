import Vue from 'vue';
import Vuex from 'vuex';
Vue.use(Vuex);

import getters from './getters.js';
import mutations from './mutations.js';
import actions from './actions.js';
import {accountModule} from '../modules/Account';

const store = new Vuex.Store({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // currently selected normalization matrix
    normalization: undefined,

    // roi settings
    roiInfo: {visible: false},

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,

    currentUser: {},

    // ion image global viewer settings
    channels: [],
    mode: 'SINGLE'
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule
  }
})

export default store;
