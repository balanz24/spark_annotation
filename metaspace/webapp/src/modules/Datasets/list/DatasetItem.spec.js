import { mount } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import router from '../../../router'
import store from '../../../store'
import { sync } from 'vuex-router-sync'
import DatasetItem from './DatasetItem.vue'
import { mockGenerateId, resetGenerateId } from '../../../../tests/utils/mockGenerateId'

Vue.use(Vuex)
sync(store, router)


describe('DatasetItem', () => {
  const user = { id: 'user' }
  const submitter = { id: 'submitter' }
  const admin = { id: 'admin', role: 'admin' }
  const dataset = {
    id: 'mockdataset',
    status: 'FINISHED',
    metadataJson: JSON.stringify({
      Metadata_Type: 'Imaging MS',
      MS_Analysis: {
        Detector_Resolving_Power: { mz: 1234, Resolving_Power: 123456 },
      },
    }),
    configJson: JSON.stringify({
      fdr: { scoring_model: null },
    }),
    databases: [
      { name: 'CHEBI', id: 2 },
      { name: 'HMDB-v2.5', id: 6 },
      { name: 'HMDB-v4', id: 22 },
    ],
    polarity: 'POSITIVE',
    fdrCounts: {
      databaseId: 6,
      dbName: 'HMDB-v2.5',
      levels: [10],
      counts: [20],
    },
    groupApproved: true,
    submitter,
    organism: 'organism',
    organismPart: 'organismPart',
    condition: 'condition',
    analyzer: {
      type: 'type'
    },
    projects: [],
    canEdit: true,
    canDelete: true,
    uploadDT: '2020-04-17T11:37:50.318667',
  }
  const unpublished = { name: 'project', publicationStatus: 'UNPUBLISHED' }
  const underReview = { name: 'project', publicationStatus: 'UNDER_REVIEW' }
  const published = { name: 'project', publicationStatus: 'PUBLISHED' }

  beforeEach(() => {
    resetGenerateId()
  })

  it('should match snapshot', () => {
    mockGenerateId(123)
    const propsData = {
      currentUser: submitter,
      dataset,
    }
    const wrapper = mount(DatasetItem, { parentComponent: { store, router }, propsData })
    expect(wrapper.element).toMatchSnapshot()
  })

  // NOTE: delete changed to dataset overview page (to be removed)
  // it('should be able to delete if unpublished', () => {
  //   const propsData = {
  //     currentUser: submitter,
  //     dataset: {
  //       ...dataset,
  //       projects: [unpublished]
  //     },
  //   }
  //   const wrapper = mount(DatasetItem, { parentComponent: { store, router }, propsData })
  //   expect(wrapper.find('.ds-delete').exists()).toBe(true)
  // })

  it('should not show the publication status if cannot edit', () => {
    const propsData = {
      currentUser: user,
      dataset: {
        ...dataset,
        canEdit: false,
        projects: [published]
      },
    }
    const wrapper = mount(DatasetItem, { parentComponent: { store, router }, propsData })
    expect(wrapper.find('.test-publication-status').exists()).toBe(false)
  })

  it('should not show the publication status if processing', () => {
    const propsData = {
      currentUser: submitter,
      dataset: {
        ...dataset,
        projects: [published],
        status: 'ANNOTATING'
      }
    }
    const wrapper = mount(DatasetItem, { parentComponent: { store, router }, propsData })
    expect(wrapper.find('.test-publication-status').exists()).toBe(false)
  })

  it('should show "Under review" status', () => {
    const propsData = {
      currentUser: submitter,
      dataset: {
        ...dataset,
        projects: [underReview]
      }
    }
    const wrapper = mount(DatasetItem, { parentComponent: { store, router }, propsData })
    expect(wrapper.find('.test-publication-status').text()).toBe('Under review')
  })

  it('should show "Published" status', () => {
    const propsData = {
      currentUser: submitter,
      dataset: {
        ...dataset,
        projects: [published]
      }
    }
    const wrapper = mount(DatasetItem, { parentComponent: { store, router }, propsData })
    expect(wrapper.find('.test-publication-status').text()).toBe('Published')
  })

  it('should prefer "Published" status', () => {
    const propsData = {
      currentUser: submitter,
      dataset: {
        ...dataset,
        projects: [published, underReview]
      }
    }
    const wrapper = mount(DatasetItem, { parentComponent: { store, router }, propsData })
    expect(wrapper.find('.test-publication-status').text()).toBe('Published')
  })

  // NOTE: options changed to dataset overview page (to be removed)
  // it('should show admin options', () => {
  //   const propsData = {
  //     currentUser: admin,
  //     dataset: {
  //       ...dataset,
  //       projects: [published]
  //     }
  //   }
  //   const wrapper = mount(DatasetItem, { parentComponent: { store, router }, propsData })
  //   expect(wrapper.find('.ds-delete').exists()).toBe(true)
  //   expect(wrapper.find('.ds-reprocess').exists()).toBe(true)
  // })
})
