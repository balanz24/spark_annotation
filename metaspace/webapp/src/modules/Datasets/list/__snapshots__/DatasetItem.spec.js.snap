// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetItem should match snapshot 1`] = `
<div
  class="dataset-item border border-solid border-gray-200 leading-5"
>
  <div
    class="opt-image--container"
  >
    <a
      class=""
      href="/datasets/mockdataset/add-optical-image"
    >
      <div
        class="thumb thumb__editable thumb__empty"
        title="Add Optical Image"
      >
        <!---->
         
        <!---->
         
        <img
          alt="Add optical image"
          class="thumb--img"
          src="../../../assets/no_opt_image.png"
        />
         
        <div
          class="thumb--overlay el-icon-edit el-icon-plus"
        />
      </div>
    </a>
  </div>
   
  <div
    class="ds-info"
  >
    <div
      class="ds-item-line flex"
    >
      <span
        class="font-bold truncate"
      />
      <mock-el-popover
        class="ml-1"
        placement="top"
        trigger="hover"
      >
        <div
          slot-key="default"
        >
          <div
            mock-v-loading="true"
          />
        </div>
        <div
          slot-key="reference"
        >
          <i
            class="el-icon-lock"
          />
        </div>
      </mock-el-popover>
      <mock-el-popover
        class="ml-1"
        placement="top"
        popper-class="min-w-0"
        trigger="hover"
      >
        <div
          slot-key="default"
        >
           
          <p
            class="leading-none m-0"
          >
            Copy dataset id to clipboard
          </p>
        </div>
        <div
          slot-key="reference"
        >
          <button
            class="button-reset flex  w-6 h-6"
          >
            <svg
              class="h-full w-full text-gray-400 hover:text-gray-500 sm-copy-icon"
              data-icon="copy-id"
            />
          </button>
        </div>
      </mock-el-popover>
    </div>
    <div
      class="ds-item-line text-gray-700"
    >
      <span
        class="ds-add-filter"
        title="Filter by species"
      >
        organism
      </span>
      , 
      <span
        class="ds-add-filter"
        title="Filter by organism part"
      >
        organismpart
      </span>
       
      <span
        class="ds-add-filter"
        title="Filter by condition"
      >
        (condition)
      </span>
    </div>
    <div
      class="ds-item-line"
    >
      <span
        class="ds-add-filter"
        title="Filter by ionisation source"
      />
       + 
      <span
        class="ds-add-filter"
        title="Filter by analyzer type"
      >
        type
      </span>
      , 
      <span
        class="ds-add-filter"
        title="Filter by polarity"
      >
        positive mode
      </span>
      , RP 123k @ 1234
    </div>
    <div
      class="ds-item-line"
    >
      Submitted 
      <mock-elapsed-time
        date="2020-04-17T11:37:50.318667"
      />
       by 
      <span
        class="ds-add-filter"
        title="Filter by submitter"
      />
    </div>
    <div
      class="ds-item-line"
    >
      <span>
        <a
          class=""
          href="/annotations?db_id=6&ds=mockdataset"
        >
          20 annotations
        </a>
         @ FDR 10% (HMDB-v2.5)
      </span>
    </div>
  </div>
   
  <div
    class="ds-actions"
  >
    <mock-el-dialog
      customclass="dataset-item-dialog"
      title="Provided metadata"
    >
      <div
        class="el-row"
      >
        <mock-el-tree
          data="[object Object],[object Object],[object Object]"
          default-expanded-keys="MS analysis,Detector resolving power,Data Management,Annotation settings"
          id="metadata-tree"
          node-key="id"
        />
      </div>
    </mock-el-dialog>
    <span>
      <i
        class="el-icon-picture"
      />
      <mock-el-popover
        placement="top"
        trigger="hover"
      >
        <div
          slot-key="default"
        >
          <div
            class="db-link-list"
          >
            Select a database:
            <div>
              <a
                class=""
                href="/annotations?db_id=2&ds=mockdataset"
              >
                CHEBI
              </a>
            </div>
            <div>
              <a
                class=""
                href="/annotations?db_id=6&ds=mockdataset"
              >
                HMDB-v2.5
              </a>
            </div>
            <div>
              <a
                class=""
                href="/annotations?db_id=22&ds=mockdataset"
              >
                HMDB-v4
              </a>
            </div>
          </div>
        </div>
        <div
          slot-key="reference"
        >
          <a>
            Browse annotations
          </a>
        </div>
      </mock-el-popover>
      <br />
    </span>
    <div>
      <i
        class="el-icon-view"
      />
      <a
        href="#"
      >
        Show full metadata
      </a>
    </div>
    <div>
      <i
        class="el-icon-data-analysis"
      />
      <a
        class=""
        classname="mr-2"
        href="/dataset/mockdataset"
      >
        <span>
          Dataset overview
        </span>
      </a>
    </div>
    <div>
      <i
        class="el-icon-edit"
      />
      <a
        class=""
        href="/datasets/edit/mockdataset"
      >
        Edit
      </a>
    </div>
  </div>
</div>
`;
